pkg/apis/duck/v1alpha1/subscribable_types_conversion.go

```diff
@@ -98,23 +103,29 @@ func (sink *SubscribableType) ConvertFrom(ctx context.Context, obj apis.Converti
 	case *duckv1beta1.Subscribable:
 		sink.ObjectMeta = source.ObjectMeta
 		sink.Status.ConvertFrom(ctx, &source.Status)
-		sink.Spec.ConvertFrom(ctx, source.Spec)
+		sink.Spec.ConvertFrom(ctx, &source.Spec)
 		return nil
 	default:
 		return fmt.Errorf("unknown version, got: %T", source)
 	}
 }
 
-// ConvertFrom helps implement apis.Convertible
-func (sink *SubscribableTypeSpec) ConvertFrom(ctx context.Context, source duckv1beta1.SubscribableSpec) {
-	if len(source.Subscribers) > 0 {
-		sink.Subscribable = &Subscribable{
-			Subscribers: make([]SubscriberSpec, len(source.Subscribers)),
-		}
-		for i, s := range source.Subscribers {
-			sink.Subscribable.Subscribers[i].ConvertFrom(ctx, s)
+// ConvertFrom implements apis.Convertible
+func (sink *SubscribableTypeSpec) ConvertFrom(ctx context.Context, obj apis.Convertible) error {
+	switch source := obj.(type) {
+	case *duckv1beta1.SubscribableSpec:
+		if len(source.Subscribers) > 0 {
+			sink.Subscribable = &Subscribable{
+				Subscribers: make([]SubscriberSpec, len(source.Subscribers)),
+			}
+			for i, s := range source.Subscribers {
+				sink.Subscribable.Subscribers[i].ConvertFrom(ctx, s)
+			}
 		}
+	default:
```

Comment:
Don't you need the conversion to v1 too here?

Reply:
I am going step by step. In this and 2 more PRs I just make things impelemtn the interface.
The v1alpha1<>v1 is to be added later in https://github.com/knative/eventing/pull/3477

Reply:
Ok ok